modules/openfast-library/src/FAST_Subs.f90

```diff
@@ -3394,6 +3521,322 @@ end subroutine cleanup
+
+      ! Passed variables
+   TYPE(FAST_ParameterType), INTENT(INOUT) :: p                               !< The parameter data for the FAST (glue-code) simulation
+   TYPE(FAST_MiscVarType),   INTENT(INOUT) :: m_FAST                          !< Miscellaneous variables
+   CHARACTER(*),             INTENT(IN)    :: InputFile                       !< Name of the file containing the primary input data
+   INTEGER(IntKi),           INTENT(OUT)   :: ErrStat                         !< Error status
+   CHARACTER(*),             INTENT(OUT)   :: ErrMsg                          !< Error message
+
+      ! Local variables:
+   INTEGER(IntKi)                :: I                                         ! loop counter
+   INTEGER(IntKi)                :: UnIn                                      ! Unit number for reading file
+   INTEGER(IntKi)                :: UnEc                                      ! I/O unit for echo file. If > 0, file is open for writing.
+   
+   REAL(ReKi)                    :: TmpAry(3)                                 ! temporary array to read in columns of case table
+
+   INTEGER(IntKi)                :: ErrStat2                                  ! Temporary Error status
+   LOGICAL                       :: Echo                                      ! Determines if an echo file should be written
+   CHARACTER(ErrMsgLen)          :: ErrMsg2                                   ! Temporary Error message
+   CHARACTER(1024)               :: PriPath                                   ! Path name of the primary file
+   CHARACTER(1024)               :: FstFile                                   ! Name of the primary ENFAST model file
+
+   CHARACTER(*),   PARAMETER     :: RoutineName = 'FAST_ReadSteadyStateFile'
+   
+
+      ! Initialize some variables:
+   UnEc = -1
+   Echo = .FALSE.                        ! Don't echo until we've read the "Echo" flag
+   CALL GetPath( InputFile, PriPath )    ! Input files will be relative to the path where the primary input file is located.
+   
+
+      ! Get an available unit number for the file.
+
+   CALL GetNewUnit( UnIn, ErrStat, ErrMsg )
+   IF ( ErrStat >= AbortErrLev ) RETURN
+
+
+      ! Open the Primary input file.
+
+   CALL OpenFInpFile ( UnIn, InputFile, ErrStat2, ErrMsg2 )
+      CALL SetErrStat( ErrStat2, ErrMsg2,ErrStat,ErrMsg,RoutineName)
+      if ( ErrStat >= AbortErrLev ) then
+         call cleanup()
+         RETURN        
+      end if
+
+
+   ! Read the lines up/including to the "Echo" simulation control variable
+   ! If echo is FALSE, don't write these lines to the echo file.
+   ! If Echo is TRUE, rewind and write on the second try.
+
+   I = 1 !set the number of times we've read the file
+   DO
+   !-------------------------- HEADER ---------------------------------------------
+
+      CALL ReadCom( UnIn, InputFile, 'File header: Version (line 1)', ErrStat2, ErrMsg2, UnEc )
+         CALL SetErrStat( ErrStat2, ErrMsg2,ErrStat,ErrMsg,RoutineName)
```

Comment:
A lot of lines could be removed with a `if (Failed())` at the end of the line :)